### PR TITLE
KEYCLOAK-7796 Expose RequestAuthenticator and BasicAuthRequestAuthenticator.getToken()

### PR DESCRIPTION
**Pull request by fschmager**
Alternative way of exposing `RequestAuthenticator` within `KeycloakAuthenticationProcessingFilter`.

#5363 solves this with inheritance, this PR here uses composition

https://issues.jboss.org/browse/KEYCLOAK-7796 

I've also tacked on exposing of `BasicAuthRequestAuthenticator#getToken` to allow overriding. _That_ was the main reason why I went through the effort of exposing `RequestAuthenticator` in the first place...

### CODE REVIEW

**Comment by sebastienblanc** [comment]
@Nowheresly What's your opinion on this approach ?

File: .gitignore [hunk 1]
```diff
@@ -8,6 +8,9 @@
 .project
 .settings
 .classpath
+bin/
+.factorypath
+
```
Comment (sebastienblanc) [inline .gitignore]: What is this about ? Is this commit needed ?

Comment (fschmager) [reply .gitignore]: "_Needed_" is strong language, but it's _nice to have_ for a dinosaur using eclipse like me :) I'd be happy to make this a separate PR is that would help though

Comment (sebastienblanc) [reply .gitignore]: Nah it's good for this small addition.

**Review by sebastienblanc — APPROVED** [review]

**Pull request merged**